@stellar/design-system-website/src/constants/details/buttons.tsx

```diff
@@ -0,0 +1,219 @@
+import { Button, Icon } from "@stellar/design-system";
+import { ComponentDetails, ComponentDetailsId } from "types/types.d";
+
+export const buttons: ComponentDetails = {
+  id: ComponentDetailsId.buttons,
+  title: "Buttons",
+  description: (
+    <p>
+      <code>Button</code> is used for actions, like in forms, while{" "}
+      <code>TextLink</code> is used for destinations, or moving from one page to
+      another.
+    </p>
+  ),
+  examples: [
+    {
+      title: "Primary (default)",
+      description: "Lorem ipsum dolor sit amet consectetur, adipisicing elit",
+      component: [
+        <Button>Label</Button>,
+        <Button disabled>Label</Button>,
+        <Button isLoading>Label</Button>,
+      ],
+    },
+    {
+      title: "Primary with icon on the left",
+      description: "",
+      component: [
```

Comment:
Maybe something for a future iteration: I wonder if there's a way to populate this examples array programmatically, as these follow a similar patter of "regular button, button disabled, button loading" 🤔 

Not a huge deal, as I imagine we probably won't be updating this site all that often

Reply:
We could probably come up with some kind of automation for these (every component will be different), but I think it's actually good to type them out. It gives us a "feeling" of how the component is used and update if anything is wrong. We won't achieve the same with abstraction. Definitely something we can improve later.